ios/FluentUI/Vnext/List/ListCell.swift

```diff
@@ -6,42 +6,108 @@
-///
-/// `title` is the first line of text, subsequently followed by `subtitle` on the second line and `footnote` on the third line.
-/// Set line limits for text using `titleLineLimit`, `subtitleLineLimit`, and `footnoteLineLimit`.
-///
-/// Any label`AccessoryView` property is a custom view at the leading/trailing end of a label, including the title, subtitle, or footnote.
-/// Currently only supports square views (width & height must be the same).
-///
-/// All `AnyView?` properties will be overwritten by its UIView equivalent. Used for SwiftUI environments.
-///
-/// `leadingView` and `trailingView` allows any custom views. Currently only supports square views (width & height must be the same).
-/// `leadingViewSize` can be specified using `MSFListCellLeadingViewSize`.
-///
-/// Set `accessoryType` using the variations specified in `MSFListAccessoryType`. This is displayed in the very trailing end of a cell.
-///
-/// `children`: define nested cells. `isExpanded` will determine if the children are initially expanded or collapsed.
-///
-/// `layoutType`:  override the default cell height.
-///
-/// `onTapAction`: perform an action on a cell tap.
-///
-/// `hasDivider`:  displays a separator beneath the cell. Default is set to `false`.
-///
-@objc public class MSFListCellState: NSObject, ObservableObject, Identifiable {
-    public var id = UUID()
-
-    @Published public var leadingView: AnyView?
-    @Published public var titleLeadingAccessoryView: AnyView?
-    @Published public var titleTrailingAccessoryView: AnyView?
-    @Published public var subtitleLeadingAccessoryView: AnyView?
-    @Published public var subtitleTrailingAccessoryView: AnyView?
-    @Published public var footnoteLeadingAccessoryView: AnyView?
-    @Published public var footnoteTrailingAccessoryView: AnyView?
-    @Published public var trailingView: AnyView?
-
-    @objc @Published public var leadingViewSize: MSFListCellLeadingViewSize = .medium {
+/// Properties that can be used to customize the appearance of the List Section.
+@objc public protocol MSFListCellState {
+    /// Custom view on the leading side of the Cell.
+    var leadingUIView: UIView? { get set }
+
+    /// Custom view on the leading side of the Cell title.
+    var titleLeadingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the trailing side of the Cell title.
+    var titleTrailingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the leading side of the Cell subtitle.
+    var subtitleLeadingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the trailing side of the Cell subtitle.
+    var subtitleTrailingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the leading side of the Cell footnote.
+    var footnoteLeadingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the trailing side of the Cell footnote.
+    var footnoteTrailingAccessoryUIView: UIView? { get set }
+
+    /// Custom view on the trailing side of the Cell.
```

Comment:
nit: how about moving the trailingUIView and the leadingViewSize closer to the leadingUIView?